src/__experimental__/components/input/input.component.js

```diff
@@ -1,99 +1,102 @@
+
+  const handleClick = (ev) => {
+    if (onClick) onClick(ev);
+    input.current.focus();
   };
 
-  handleFocus = (ev) => {
-    if (this.props.onFocus) this.props.onFocus(ev);
-    if (this.context && this.context.onFocus) this.context.onFocus(ev);
-    if (this.props.type === 'text') selectTextOnFocus(this.input);
+  const handleFocus = (ev) => {
+    if (onFocus) onFocus(ev);
+    if (context.onFocus) context.onFocus(ev);
+    if (groupContext.onFocus) groupContext.onFocus(ev);
+    if (type === 'text') selectTextOnFocus(input);
   };
 
-  handleBlur = (ev) => {
-    if (this.props.onBlur) this.props.onBlur(ev);
-    if (this.context && this.context.onBlur) this.context.onBlur(ev);
+  const handleBlur = (ev) => {
+    if (onBlur) onBlur(ev);
+    if (context.onBlur) context.onBlur(ev);
+    if (groupContext.onBlur) groupContext.onBlur(ev);
   };
 
-  handleChange = (ev) => {
-    if (this.props.onChange) {
-      this.props.onChange(ev);
+  const handleDeferred = ({ currentTarget, target }) => {
+    if (onChangeDeferred) {
+      clearTimeout(deferredTimeout.current);
+      deferredTimeout.current = setTimeout(() => {
+        onChangeDeferred({ currentTarget, target });
+      }, (deferTimeout || 750));
     }
+  };
 
-    this.handleDeferred(ev);
-  }
-
-  handleDeferred = ({ currentTarget, target }) => {
-    if (this.props.onChangeDeferred) {
-      clearTimeout(this.deferredTimeout);
-      this.deferredTimeout = setTimeout(() => {
-        this.props.onChangeDeferred({ currentTarget, target });
-      }, (this.props.deferTimeout || 750));
-    }
-  }
-
-  render() {
-    const {
-      inputRef,
-      onChangeDeferred,
-      ...props
-    } = this.props;
-    const eventHandlers = {
-      onFocus: this.handleFocus,
-      onBlur: this.handleBlur,
-      onClick: this.handleClick,
-      onChange: this.handleChange
-    };
+  const handleChange = (ev) => {
+    if (onChange) onChange(ev);
+    handleDeferred(ev);
+  };
 
-    return (
-      <StyledInput
-        { ...props }
-        id={ this.props.id || this.props.name }
-        ref={ this.input }
-        data-element='input'
-        { ...eventHandlers }
-      />
-    );
-  }
-}
+  return (
+    <StyledInput
+      { ...rest }
+      name={ name }
+      type={ type }
+      id={ id || name }
+      ref={ input }
+      data-element='input'
+      onFocus={ handleFocus }
+      onBlur={ handleBlur }
+      onClick={ handleClick }
+      onChange={ handleChange }
+    />
+  );
+});
+
+Input.propTypes = {
+  className: PropTypes.string,
+  id: PropTypes.string,
+  inputRef: PropTypes.func, // a callback to retrieve the input reference
+  name: PropTypes.string,
+  onBlur: PropTypes.func,
+  onClick: PropTypes.func,
+  onKeyDown: PropTypes.func,
+  onFocus: PropTypes.func,
+  autoFocus: PropTypes.bool,
+  onChange: PropTypes.func,
+  onChangeDeferred: PropTypes.func,
+  deferTimeout: PropTypes.number,
+  type: PropTypes.string
+};
 
 Input.defaultProps = {
```

Comment:
Can we move these default values into the destuctured props at the top?

```
const Input = React.forwardRef(({
  ...,
  type = 'text',
  ...
}, ref) => {
```

https://github.com/reactjs/rfcs/blob/3416f0de38ee120f6d9dac7f7c9405b77cf37bdc/text/0000-create-element-changes.md#deprecate-defaultprops-on-function-components